### PR TITLE
feat: change separator color in `OverviewBlock` component

### PR DESCRIPTION
**Pull request by pudek357**
[SPT-1788](https://toptal-core.atlassian.net/browse/SPT-1788)

### Description
As per feedback, we would like to adjust on Staff Portal dashboard page color of the vertical separator in OverviewBlock.
Discussion on Design channel on slack [here](https://toptal-core.slack.com/archives/CM8KE183V/p1626277935236400) and [here](https://toptal-core.slack.com/archives/CCC3GP6CC/p1627284635023100).

Abstract updated [here](https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/550eaf1f40f66ec5bcc307ea06962f0eb0f7fbd3/files/4ba86227-d49e-435b-92d3-1b41a4b9531a/layers/3B863FAB-C94C-4EE2-8061-FBFAB45778AE?mode=design&sha=550eaf1f40f66ec5bcc307ea06962f0eb0f7fbd3).

### Screenshots
| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|  |  |

### Review
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Comment by pudek357** [comment]
@toptal-bot run package:alpha-release

**Comment by pudek357** [comment]
@toptal-bot run package:alpha-release

**Comment by pudek357** [comment]
@toptal-anvil ping reviewers

File: packages/picasso-lab/src/OverviewBlock/OverviewBlock.tsx [hunk 1]
```diff
@@ -49,6 +49,8 @@ export type Props = BaseProps &
     label: string
     /** The color variant  */
     variant?: Variant
+    /** Sets darker color for vertical separator */
+    hasDarkerVerticalSeparator?: boolean
```
Comment (teimurjan) [inline packages/picasso-lab/src/OverviewBlock/OverviewBlock.tsx:53]: This name looks long and unclear to me. What do you think about making such a prop?

```suggestion
    separatorVariant?: 'default' | 'dark'
```

File: packages/picasso-lab/src/OverviewBlock/story/DarkerVerticalSeparator.example.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { OverviewBlock } from '@toptal/picasso-lab'
+
+const DarkerVerticalSeparator = () => {
+  return (
+    <OverviewBlock.Group>
+      <OverviewBlock hasDarkerVerticalSeparator value='4249' label='ACH' />
```
Comment (teimurjan) [inline packages/picasso-lab/src/OverviewBlock/story/DarkerVerticalSeparator.example.tsx:7]: I think this prop should be passed to `OverviewBlock.Group` and provided via context just like `blockWidth` (https://github.com/toptal/picasso/blob/056761c706d5945916499c85242244570b8cd551/packages/picasso-lab/src/OverviewBlockGroup/OverviewBlockGroup.tsx#L38)

**Review by teimurjan — CHANGES_REQUESTED** [review]

**Comment by denieler** [comment]
Has this change being discussed with the `#base-core` channel/designers?

**Comment by pudek357** [comment]
@denieler in **Description**, you have a link to the thread in Slack.

**Comment by teimurjan** [comment]
> Has this change being discussed with the `#base-core` channel/designers?

https://toptal-core.slack.com/archives/CM8KE183V/p1626279193238100?thread_ts=1626277935.236400&cid=CM8KE183V

**Comment by denieler** [comment]
thank you @pudek357 @teimurjan 🙇 I've read that slack thread, but not that far.

@pudek357 did designers change the design in Abstract in BASE project?

**Comment by denieler** [comment]
https://share.goabstract.com/39332544-280f-4103-8908-b04220d5932b?mode=design&sha=b17f1de5aad308ae26d90e16bc0d7073d4330b68

It looks like it's not. Please ask `#base-core` to update the design 🙇‍♂️

File: packages/picasso-lab/src/OverviewBlock/OverviewBlock.tsx (hunk 1 above)
Comment (denieler) [reply packages/picasso-lab/src/OverviewBlock/OverviewBlock.tsx:53]: do we really need 2 types of the separator? It looks like they just asked to change the color. Am I missing something?

Comment (pudek357) [reply packages/picasso-lab/src/OverviewBlock/OverviewBlock.tsx]: We don't need, via: [Slack](https://toptal-core.slack.com/archives/CCC3GP6CC/p1627379282026500?thread_ts=1627284635.023100&cid=CCC3GP6CC).

**Comment by pudek357** [comment]
@denieler abstract has been updated [here](https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/550eaf1f40f66ec5bcc307ea06962f0eb0f7fbd3/files/4ba86227-d49e-435b-92d3-1b41a4b9531a/layers/3B863FAB-C94C-4EE2-8061-FBFAB45778AE?mode=design&sha=550eaf1f40f66ec5bcc307ea06962f0eb0f7fbd3).

**Review by denieler — APPROVED** [review]

**Review by teimurjan — APPROVED** [review]

**Pull request merged**